src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -338,11 +328,18 @@ class JesBackend extends Backend with LazyLogging with ProductionJesAuthenticati
     val initializedStatus = run.waitUntilRunningOrComplete
     backendCall.callAbortRegistrationFunction.register(AbortFunction(() => run.abort()))
 
-    def wdlFileToGcsPath(value: WdlValue, coerceTo: WdlType) =
-      if (coerceTo == WdlFileType && WdlFileType.isCoerceableFrom(value.wdlType))
-        jesOutputs find { _.name == value.valueString } map { j => WdlFile(j.gcs) } getOrElse value
-      else
-        value
+    def wdlValueToGcsPath(value: WdlValue): WdlValue = {
+      def toGcsPath(wdlFile: WdlFile) = jesOutputs collectFirst { case o if o.name == wdlFile.valueString => WdlFile(o.gcs) } getOrElse value
+
+      value match {
+        case wdlArray: WdlArray => wdlArray map wdlValueToGcsPath
+        case wdlMap: WdlMap => wdlMap map {
+          case (k, v) => wdlValueToGcsPath(k) -> wdlValueToGcsPath(v)
+        }
+        case file: WdlFile => toGcsPath(file)
+        case other => other
```

Comment:
This is not unit tested at all AFAIK. But I'll see if I can extract it to unit test it. In any case this code path is tested via Tyburn for sure.